lib/web/apiserver.go

```diff
@@ -2500,19 +2500,18 @@ func (h *Handler) WithJavascriptRedirect(fn redirectHandlerFunc) httprouter.Hand
 			redirectURL = client.LoginFailedRedirectURL
 		}
 
-		const js = `
+		const html = `
```

Comment:
Instead of using `Fprintf`, we should use package `html/template` here, which is safe against code injection.

Reply:
you can also remove `nonce, err := utils.CryptoRandomHex(auth.TokenLenBytes)` as this is unnecessary after the script tag has been removed.